### DISCUSSION TITLE
VOI stat

### DISCUSSION THREAD

**Discussion by rvboutin** [brainvisa/brainvisa.github.io#87] [Q&A]
Hi,
I use Brainvisa/Anatomist to analyse preclinical PET/SPECT-CT and MR scans, and I cannot find a toolbox to quantify VOI, which seems strange, I must have missed something...
If not, the AimsVoiStat to quantify ROI on images (mostly useful for PET and SPECT) is still in the binary, but there is no toolbox in Brainvisa to interface with that command. Would it be possible to add this command as toolbox in Brainvisa?
Thanks,
Hervé

**Comment by sapetnioc** [comment]
I did not find a generic BrainVISA process for this. But I found 94 processes using `AimsRoiFeatures` which is the replacement for  `AimsVOIStat` (obsolete since several years). Most of these processes are in toolboxes that are not in the BrainVISA release (because of their authors decision). The difficulty is that they all do something specific to the context of the process (for instance to facilitate ROI selection or to do a few processing on output such as unit conversion).

What would be the goal of such process for you ? Just having a GUI to select files and parameters or more ? In your PET/SPECT case, I suspect that a few specific code would be needed too (if I remember well `AimsVoiStat` used to do some units conversion but `AimsRoiFeatures` is completely generic). To be honest, it may not be easy to find someone willing to develop and maintain such a tool. But it is not a big deal to do it for someone knowing Python and a bit of BrainVISA. I can help to start if you have someone in mind.

**Comment by rvboutin** [comment]
Hi,
I just need a GUI tool really to input all the required files (volume = dynamic or static PET scan, Arg=ROI files and output text file) for the AimsVoiStat command. I guess I can try to modify the python code of an existing toolbox to get this works. I do not know the AimsRoiFeatures command, I never used it. AimsVoiStat does not do anything fancy like unit conversion as far as I am aware, it just quantifies a 4D volume with a given ROI file and spit out a result file as text "|" delimited file which I can easily import in Excel (I have created Macros to input injected dose, etc.).
I would prefer to have a GUI toolbox because to be honest command line implementation is not very user friendly when I train students and post-docs, and it is prone to errors (typo, etc.). even with drag & drop of files names.
The PET/preclinical aspect of Brainvisa has been abandoned and it's really a shame because the capabilities of Brainvisa (database, etc.) are really great for preclinical imaging and could be use by many more users in the field of preclinical research and further improved. Alternative software like PMOD or VivoQuant are super expensive and do not offer superior capabilities (or very few) over Brainvisa. There is a massive untapped pool of potential user here.
Many tools are already there, but they are lacking GUI toolbox.
Thanks,
Hervé

**Reply by sapetnioc** [reply]
Well, you should have a look at `AimsRoiFeatures` since it is, to my biased opinion, simpler and easier to use than `AimsVoiStat`. For instance, it produces CSV files that can be directly imported in LibreOffice or Excel. 

There is still a toolbox for Nuclear imaging using BrainVISA but it is developed by  [CATI](https://cati-neuroimaging.com) for internal use (quality control of human brain PET/SPECT images)  and is so connected to its internal infrastructure (spread over three sites) that it cannot be distributed for individual usage. However, I may have found exactly the kind of process you need in that toolbox (using `AimsRoiFeature`though). I will get in touch with the authors and me or them will come back to you.

**Comment by Hboni** [comment]
In the nuclear medicine toolbox that we use and that I am in charge to maintain, we have a process that seems to suit your needs. It allows launching `AimsROIFeatures` to make quantification in VOI. 
As this process is only used in the nuclear medicine toolbox, we kept it in this toolbox, but as it is not specific to the modality, we may move it to a more generic toolbox.

**Comment by sapetnioc** [comment]
And for your second question about Anatomist. You can launch `anatomist` then load a file. That will create an object corresponding to that file (image, mesh, etc.). Then, to view the object you need to create a view window and to put the object in it. More info about that in [Anatomist tutorial](https://brainvisa.info/anatomist-5.0/user_doc/anatomist_tutorial.html).

You may be interested in running `anasimpleviewer.py` which has been first developed as a demo Python script for creating a custom GUI with Anatomist views. But we kept it because it offer a much simpler GUI (with a very small subset of Anatomist features). More info on the tutorial page: https://brainvisa.info/anatomist-5.0/user_doc/anatomist_tutorial.html#anasimpleviewer-a-simplified-version-of-anatomist
